lib/PN5180/PN5180.cpp

```
@@ -474,6 +474,15 @@ bool PN5180::setRF_on() {
   {
     unsigned long t = millis();
     while (0 == (TX_RFON_IRQ_STAT & getIRQStatus())) { // wait for RF field to set up
+      // TX_RFON is a TRANSITION interrupt — commanding RF on while the field
+      // is already energized produces no edge and no IRQ. Check the live
+      // field state before calling it a failure: an already-on field is
+      // success (this fired as a false "TIMEOUT TX_RFON_IRQ" on every idle
+      // scan cycle once the timeout logs became always-on).
+      uint32_t rfStatus = 0;
+      if (readRegister(RF_STATUS, &rfStatus) && (rfStatus & (1UL << 20))) {
```

Clang (14.0.6)

[warning] implicit conversion 'unsigned long' -> bool

(readability-implicit-bool-conversion)